### PR TITLE
Adding Oracle Cloud Storage as a backup provider

### LINKED ISSUE #1569: Support for OCI Object Storage as a Backup Provider
Adding a document to describe how to use S3-compatible OCI Object Storage as a backup provider for Velero.

### PR DESCRIPTION
**Pull request by mickeyboxell**
fixes  #1569 

Adding documentation to use OCI Object Storage as an S3-compatible backup provider. 

The plugin to provide volume snapshot and object storage capabilities is currently under development. This document will be updated upon it's completion.

### CODE REVIEW

File: site/docs/master/oracle-config.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+   $ velero backup get 
+   NAME     STATUS      CREATED     EXPIRES     STORAGE     LOCATION        SELECTOR
+   
+    ```
+
+
+
+## Additional Reading 
+
+* [Official Velero Documentation](https://heptio.github.io/velero/master/)
```
Comment (carlisia) [inline site/docs/master/oracle-config.md]: Please point to the docs on our website instead: https://velero.io/docs/master/

File: site/docs/master/oracle-config.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
+# Use Oracle Cloud as a Backup Storage Provider for Velero
+
+## Introduction
+
+[Velero](https://heptio.github.io/velero/master/) is a tool used to backup and migrate Kubernetes applications. We support [Oracle Cloud Object Storage](https://docs.cloud.oracle.com/iaas/Content/Object/Concepts/objectstorageoverview.htm) as a destination for Velero backups.
```
Comment (carlisia) [inline site/docs/master/oracle-config.md]: Unfortunately, we can't support and test storage providers beyond the ones we already do. The `We support` language needs to be changed.

File: site/docs/master/oracle-config.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
+# Use Oracle Cloud as a Backup Storage Provider for Velero
+
+## Introduction
+
+[Velero](https://heptio.github.io/velero/master/) is a tool used to backup and migrate Kubernetes applications. Here are the steps to use [Oracle Cloud Object Storage](https://docs.cloud.oracle.com/iaas/Content/Object/Concepts/objectstorageoverview.htm) as a destination for Velero backups.
```
Comment (skriss) [inline site/docs/master/oracle-config.md]: Given that this page will already be on the website, I'm not sure it needs to be a link, but if you want to leave the link, let's set this to just https://velero.io

File: site/docs/master/oracle-config.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
+# Use Oracle Cloud as a Backup Storage Provider for Velero
+
+## Introduction
+
+[Velero](https://heptio.github.io/velero/master/) is a tool used to backup and migrate Kubernetes applications. Here are the steps to use [Oracle Cloud Object Storage](https://docs.cloud.oracle.com/iaas/Content/Object/Concepts/objectstorageoverview.htm) as a destination for Velero backups. 
+
+1. [Download Velero](README.md#download-velero)
```
Comment (skriss) [inline site/docs/master/oracle-config.md]: I think you want to drop the `README.md` from all of the links in this TOC

File: site/docs/master/oracle-config.md [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+   $ velero backup delete nginx-backup
+   Are you sure you want to continue (Y/N)? Y
+   Request to delete backup "nginx-backup" submitted successfully.
+   The backup will be fully deleted after all associated data (disk snapshots, backup files, restores) are removed.
+   ```
+
+   This asks the Velero server to delete all backup data associated with `BACKUP_NAME`. You need to do this for each backup you want to permanently delete. A future version of Velero will allow you to delete multiple backups by name or label selector.
+
+   Once fully removed, the backup is no longer visible when you run: `velero backup get BACKUP_NAME` or more generally `velero backup get`:
+    ```
```
Comment (skriss) [inline site/docs/master/oracle-config.md]: you need one blank line above this for the code block to be rendered properly

**Comment by mickeyboxell** [comment]
@skriss I made your recommended changes. Is there anything else for me to do on my end? Re: contact info - I couldn't find example from the other providers, so I left this off the documentation for now.

File: site/docs/master/oracle-config.md [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+   Once fully removed, the backup is no longer visible when you run: `velero backup get BACKUP_NAME` or more generally `velero backup get`:
+   
+   ```
+   $ velero backup get nginx-backup
+   An error occurred: backups.velero.io "nginx-backup" not found
+
+   $ velero backup get 
+   NAME     STATUS      CREATED     EXPIRES     STORAGE     LOCATION        SELECTOR
+   
+    ```
```
Comment (skriss) [inline site/docs/master/oracle-config.md]: Looks like maybe this has one leading space too many, it's not being rendered properly - see https://deploy-preview-1570--velero.netlify.com/docs/master/oracle-config/

**Comment by skriss** [comment]
@mickeyboxell if you don't want to add contact info that's OK -- just thought you might want users to have a person to contact if they had any issues.

The one other thing you might want to do is make this same set of changes in the `site/docs/v1.0.0` directory so they're immediately reflected in the v1.0.0-tagged docs. Not required, but that's the default docs tag that's shown on the website.

File: site/docs/master/oracle-config.md (hunk 6 above)
Comment (skriss) [reply site/docs/master/oracle-config.md]: Just to clarify, the line with the 3 backticks to close the code block (L236 in the most recent version) is the one with one too many leading spaces.

**Comment by skriss** [comment]
@mickeyboxell see https://github.com/heptio/velero/pull/1570#discussion_r295394981 (the line with the extra space is now L238)

File: site/docs/master/oracle-config.md (hunk 6 above)
Comment (mickeyboxell) [reply site/docs/master/oracle-config.md]: Thanks for following up! I just caught it and I'll update one more time.

Comment (mickeyboxell) [reply site/docs/master/oracle-config.md]: Fingers crossed 🤞

**Comment by mickeyboxell** [comment]
Is there a recommended format for adding contact information?

File: site/docs/master/oracle-config.md (hunk 6 above)
Comment (skriss) [reply site/docs/master/oracle-config.md]: woohoo looks good on the preview!

**Comment by skriss** [comment]
@mickeyboxell just checking to see if you were going to (a) add contact info and (b) copy this page into the `site/docs/v1.0.0` directory. Ready to merge when that's done, or if you confirm you aren't planning to do those!

**Comment by mickeyboxell** [comment]
@skriss I'll skip the contact info for now and add it when we update the docs with the volume snapshot provider version. I've added the updates to v1.0.0. We should be good to merge. 😄

**Pull request merged**